File: src/Document.cpp
#include "Document.h"

void Document::save(fs::path parent_dir){
	Ressource::save(parent_dir);
    cout<<location(parent_dir).c_str()<<endl;
	ofstream ofs( location(parent_dir).c_str(), ofstream::binary );
	ofs<<data;
	ofs.close();
}

File: src/Document.py
from .Ressource import Ressource, RType
import shutil
import os

class Document( Ressource ):
	_type	= RType.document
	def __init__(self, i=-1, tmp=None): 
		Ressource.__init__(self, i, tmp)
		
	def save(self, parent_dir, closed=True):
		"""
		"""
		Ressource.save(self, parent_dir)
		
		if not self._tmp :
			return
			
		with open( os.path.join(self._location, "data"), 'wb') as newFile:
			self._tmp.seek(0) 
			shutil.copyfileobj(self._tmp, newFile, 32768)
		
		if closed:
			self._tmp.close()
